compiler/test/fail_compilation/b4541.d

```diff
@@ -0,0 +1,11 @@
+/*
+TEST_OUTPUT:
+---
+fail_compilation/b4541.d(10): Error: cannot take address of intrinsic function `sin`
+fail_compilation/b4541.d(10):        use `&std.math.sin` instead
+---
+*/
+import core.math;
+void test() {
+    real function(real) c = &sin;
+}
```

Comment:
This test works with gdc though, albeit I'm going to be a bit sneaky about it.

Looking at this PR made me notice that gdc is not expanding referenced intrinsics in the same way as directly called intrinsics are.  So I'm going to commit to mainline a change to right that wrong, and allow the gcc back-end to automatically swap `core.math.sin` for `sinl`.

Reply:
I know this can work on particular cases but for example llvm.stacksave llvm.stackrestore are meant to inject instructions inline, that why I mentioned that this PR is "the only simple solution." (to solve the issue for good).